### PR TITLE
Feature/nfct

### PR DESCRIPTION
**Pull request by mischmi96**
Adding an interface for the NFCT in Julia.

### CODE REVIEW

**Comment by tvolkmer** [comment]
Should we build the julia interfaces when `--enable-all` is set?

File: julia/nfct/NFCT.jl [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+# FFTW flags
+FFTW_MEASURE = UInt32(0)
+FFTW_DESTROY_INPUT = UInt32(1)<<0
+FFTW_UNALIGNED = UInt32(1)<<1
+FFTW_CONSERVE_MEMORY = UInt32(1)<<2
+FFTW_EXHAUSTIVE = UInt32(1)<<3
+FFTW_PRESERVE_INPUT = UInt32(1)<<4
+FFTW_PATIENT = UInt32(1)<<5
+FFTW_ESTIMATE = UInt32(1)<<6
+FFTW_WISDOM_ONLY = UInt32(1)<<21
```
Comment (michaelquellmalz) [inline julia/nfct/NFCT.jl]: Does it make sense to use this flag (or some others like FFTW_CONSERVE_MEMORY , FFTW_PRESERVE_INPUT)

**Comment by mischmi96** [comment]
@tvolkmer bump

File: julia/nfct/Makefile.am [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  libadd_for_fftw_threads=@fftw3_threads_LIBS@
+else
+  libadd_for_fftw_threads=
+endif
+
+libnfctjulia_la_LIBADD = $(top_builddir)/libnfft3@PREC_SUFFIX@_julia.la @fftw3_LDFLAGS@ $(libadd_for_fftw_threads) @fftw3_LIBS@
+
+libnfctjulia_la_LDFLAGS = -no-undefined -module -shared -avoid-version @fftw3_LDFLAGS@
+
+EXTRA_DIST = NFCT.jl #simple_test_1d.jl simple_test_2d.jl simple_test_3d.jl
```
Comment (michaelquellmalz) [inline julia/nfct/Makefile.am]: why is simple_test_xd commented out?

**Pull request merged**